test/views/agent/DeductionBreakdownViewSpec.scala

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.agent
+
+import assets.MessagesLookUp.DeductionBreakdown
+import assets.CalcBreakdownTestConstants
+import config.FrontendAppConfig
+import enums.Estimate
+import models.calculation.CalcDisplayModel
+import org.jsoup.nodes.Element
+import testUtils.ViewSpec
+import views.html.agent.DeductionBreakdown
+
+class DeductionBreakdownViewSpec extends ViewSpec {
+
+  lazy val mockAppConfig: FrontendAppConfig = app.injector.instanceOf[FrontendAppConfig]
+
+  val deductionBreakdown: DeductionBreakdown = app.injector.instanceOf[DeductionBreakdown]
+  val taxYear = 2017
+
+  class DeductionBreakdownSetup() extends Setup(
+    deductionBreakdown( CalcDisplayModel("", 1,
+      CalcBreakdownTestConstants.calculationNoBillModel,
+      Estimate),taxYear,"testBackURL")
+  )
+
+
+
+  val backUrl = "/report-quarterly/income-and-expenses/view/calculation/2021"
+
+  "The deduction breakdown view" when {
+
+    "provided with a calculation without tax deductions for the 2017 tax year" should {
+      val taxYear = 2017
+
+
+
+      "have the correct title" in new DeductionBreakdownSetup {
+        document title() shouldBe DeductionBreakdown.agentTitle
+      }
+
+      "have the correct heading" in new DeductionBreakdownSetup {
+        content hasPageHeading DeductionBreakdown.heading(taxYear)
+        content.h1.select(".heading-secondary").text() shouldBe DeductionBreakdown.subHeading(taxYear)
+      }
+
+      "have the correct guidance" in new DeductionBreakdownSetup {
+        val guidance: Element = content.select("p").get(1)
+        guidance.text() shouldBe DeductionBreakdown.guidance(taxYear)
+      }
+
+      "have an deduction table" which {
+
+        "has only one table row" in new DeductionBreakdownSetup {
+          content hasTableWithCorrectSize (1, 1)
+        }
+
+        "has a total line with a zero value" in new DeductionBreakdownSetup {
+          val row: Element = content.table().select("tr").first()
+          row.select("td").first().text() shouldBe DeductionBreakdown.total
+          row.select("td").last().text() shouldBe "£0.00"
+        }
+      }
+    }
+
+    "provided with a calculation with all tax deductions for the 2018 tax year" should {
+      val taxYear = 2018
+
+      class DeductionBreakdownSetup2() extends Setup(
```

Comment:
Optional readability note: the `DeductionBreakdownSetup2 ` can be renamed to `DeductionBreakdownSetup2018`